modules/core/src/bitgo.ts

```diff
@@ -48,63 +49,42 @@ if (!(process as any).browser) {
   require('superagent-proxy')(superagent);
 }
 
-// Patch superagent to return bluebird promises
-const _end = (superagent as any).Request.prototype.end;
-(superagent as any).Request.prototype.end = function (cb) {
-  const self = this;
-  if (typeof cb === 'function') {
-    return _end.call(self, cb);
-  }
-
-  return new Bluebird.Promise(function (resolve, reject) {
-    let error;
-    try {
-      return _end.call(self, function (error, response) {
-        if (error) {
-          return reject(error);
-        }
-        return resolve(response);
-      });
-    } catch (_error) {
-      error = _error;
-      return reject(error);
-    }
-  });
-};
-
 // Handle HTTP errors appropriately, returning the result body, or a named
 // field from the body, if the optionalField parameter is provided.
-(superagent as any).Request.prototype.result = function (optionalField?: string) {
+function toBitgoRequest<ResponseResultType = any>(req: superagent.SuperAgentRequest): BitGoRequest<ResponseResultType> {
```

Comment:
This could maybe have some more comments about what it's doing and why it's needed, even just taking a line or two from the PR description and adding it here.

Reply:
yep, that's worthwhile